src/pages/chat/chat.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useEffect, useRef, useState } from "react";
+import styled from "styled-components";
+import ChatInput from "../../components/ChatInput/chatinput";
+import userData from "../../assets/datas/userdata.json";
+import chatData from "../../assets/datas/chatdata.json";
+import ChatTitle from "../../components/ChatTitle/chatTitle";
+import MessageList from "../../components/Message/messageList";
+import Message from "../../components/Message/message";
+import "../../styles/colors.css";
+import bottomBar from "../../assets/images/LightBottomBar.svg";
+import StatusBar from "../../components/StatusBar/statusbar";
+
+interface Message {
+  id: number;
+  sender: string;
+  content: string;
+  showIcon: boolean;
+}
+
+function Chat() {
+  const savedChats = JSON.parse(
+    localStorage.getItem("chatMessages") || JSON.stringify(chatData)
+  );
+
+  const [messages, setMessages] = useState<Message[]>(savedChats);
+  const [messageId, setMessageId] = useState<number>(chatData.length);
+  const [nowUser, setNowUser] = useState(userData.users[0]);
+
+  const handleSendMessage = (content: string) => {
+    const newMessage: Message = {
+      id: messageId,
+      sender: nowUser.name,
+      content,
+      showIcon: false,
+    };
+
+    // 메세지가 추가됐을 때 가장 마지막 index의 메세지와 비교하여 showIcon 업데이트
+    if (
+      messages.length > 0 &&
+      messages[messages.length - 1].sender !== nowUser.name
+    ) {
+      messages[messages.length - 1].showIcon = true;
+    }
+
+    const updatedMessages = [...messages, newMessage];
+    setMessages(updatedMessages);
+
+    // 다음 메시지를 위해 messageId 업데이트
+    setMessageId(messageId + 1);
+    // 로컬 스토리지에 채팅 데이터 저장
+    localStorage.setItem("chatMessages", JSON.stringify(updatedMessages));
+  };
+
+  const changeUser = () => {
+    setNowUser((prev) =>
+      prev.name === userData.users[0].name
+        ? userData.users[1]
+        : userData.users[0]
+    );
+
+    //사용자 바뀌면, 마지막 상대방 메시지 showIcon 처리해주는 부분
+    setMessages((prev) => {
+      const lastMessageIndex = prev.length - 1;
+      //userChange 동시에 여러번 누르면 아이콘 동시에 뜨는것 방지하기 위해
+      if (lastMessageIndex >= 1) {
+        if (
+          prev[lastMessageIndex - 1].sender === prev[lastMessageIndex].sender &&
+          prev[lastMessageIndex - 1].showIcon
+        ) {
+          prev[lastMessageIndex - 1].showIcon = false;
+        }
+        prev[lastMessageIndex].showIcon = true;
+      }
```

Comment:
```suggestion
  if (
    lastMessageIndex >= 1 &&
    prev[lastMessageIndex - 1].sender === prev[lastMessageIndex].sender &&
    prev[lastMessageIndex - 1].showIcon
  ) {
    prev[lastMessageIndex - 1].showIcon = false;
  }
  prev[lastMessageIndex].showIcon = true;
```

위 방법도 좋지만 이중 if문 사용없이 마지막 요소의 showIcon을 항상 true로 두면 조금 더 깔끔한 코드가 될 것 같아요!